### ISSUE TITLE
Restore after save state fails because of direction being serialized as an object

### ISSUE THREAD

**Issue by Julesdevops** [Kitware/glance#480]
Seems like `direction` is stored as a `Float64Array`, `JSON.stringify` serializes it as an JS object :(
Therefore when restoring we have a failure since it tries to feed an object into `macro.setArray`
I'll fix it

**Comment by floryst** [comment]
Good catch! Do you anticipate this being a fix into vtk.js or Glance? `vtkObj.toJSON()` implies the output should be JSON-serializable, but right now it definitely is not.

**Comment by Julesdevops** [comment]
yeah I guess `vtkObj.toJSON` is broken right now? did not think about that.
And we may want to fix `getState` as well since it should return a representation that should be directly JSON-serializable (e.g., returning plain JS array instead of typed array where applicable)

**Comment by Julesdevops** [comment]
we may have nothing to fix here in Glance indeed
